### ISSUE TITLE
[Bug][Pager] Dynamic height pager based on items' height

### ISSUE THREAD

**Issue by tuanchauict** [google/accompanist#413]
## Describe the bug
The height of `HorizontalPager` does not match with the value which is set to its modifier after scrolling back and forth several times

## To Reproduce
```kt
@Composable
fun Pager() {
    val heights = listOf(
        100.dp,
        150.dp,
        100.dp,
        150.dp,
        100.dp,
        150.dp
    )
    val pagerState = rememberPagerState(pageCount = heights.size)
    HorizontalPager(
        state = pagerState,
        modifier = Modifier.height(height = heights[pagerState.currentPage]) // <--- dynamic height
    ) { page ->
        Surface(color = Color.Yellow) {
            Text(
                text = "Page: $page",
                modifier = Modifier
                    .fillMaxWidth()
                    .height(height = heights[page])
            )
        }
    }
}
```

## Expected behavior
The pager's height is correct to the current item's height.

## Screenshots?
<img src="https://user-images.githubusercontent.com/955306/118414435-80374480-b6df-11eb-82c9-5c8116510a9a.gif" height="400"/>

## Environment:
 - Android OS version: Android 11
 - Device: Android Studio's design preview, Pixel 3a
 - Accompanist version: `accompanist-pager:0.9.1`

**Comment by chrisbanes** [comment]
This is working as intended. You're setting the `height()` for both the pager and the content, and they happen at different times.

If you really want this UX (which is pretty bad imo), set the height of the `HorizontalPager` only:

``` kotlin
HorizontalPager(
    ...
    modifier = Modifier.height(height = heights[pagerState.currentPage])
) { page ->
    // fillMaxSize() so that the item fills the Pager
    Surface(Modifier.fillMaxSize()) {
        // ...
    }
}
```

**Comment by tuanchauict** [comment]
Mmmm.... it seems not to work as intended when we swipe left (swiping right is ok)
<img src="https://user-images.githubusercontent.com/955306/118469641-d9839000-b740-11eb-97fb-dd157be4ead4.gif" height="300"/>

Btw, in my case, the height of each page is determined by its content, not predetermined like the code. 

(I agree with you this is really bad UX too)

**Comment by chrisbanes** [comment]
> it seems not to work as intended when we swipe left (swiping right is ok)

This is due to a bug in `0.9.1` which is fixed by #400. Will be out in the next release this week (or use the [snapshots](https://google.github.io/accompanist/using-snapshot-version/) for now).

> Btw, in my case, the height of each page is determined by its content, not predetermined like the code.

`HorizontalPager` will wrap the contents height. Content ==  current page + any offscreen pages.
